app/src/main/java/org/mozilla/focus/ext/SessionState.kt

```diff
@@ -23,3 +25,18 @@ fun SessionState.ifCustomTab(): CustomTabSessionState? {
 fun SessionState.isCustomTab(): Boolean {
     return this is CustomTabSessionState
 }
+
+/**
+ * Removes a [TrackingProtectionException] if one is set for the current session.
+ */
+fun SessionState.removeFromExceptions(trackingProtectionUseCases: TrackingProtectionUseCases) {
+    var exceptions: List<TrackingProtectionException>
+    trackingProtectionUseCases.fetchExceptions.invoke {
+        exceptions = it.filter { exception ->
+            exception.url == this.content.url
+        }
+        exceptions.firstOrNull()?.let {
+            trackingProtectionUseCases.removeException(exceptions.first())
```

Comment:
Not sure why and where if passing the tab `id` the exception is not removed by using the other `removeException` method that has an `id` param. So I used this approach to identify the exception that needs removal. Question: Can this have multiple exceptions return, should we iterate through them instead of just removing the first one?

Reply:
Do we still need this the AC [fix](https://github.com/mozilla-mobile/android-components/pull/10728)? Is the issue still reproducible?

Reply:
The change is still not in Focus. I will keep this until then, I think we will still need some changes in the migration middleware.

Reply:
👍🏽